ui/src/app/views/app/component.ts

```diff
@@ -95,7 +95,7 @@ export class AppComponent implements OnInit {
     this.user.subscribe(user => {
       this.isSignedIn = true;
       this.router.navigated = false;
-      this.router.navigateByUrl(this.router.url);
+      this.router.navigateByUrl(window.location.pathname);
```

Comment:
I believe this works because you have an `ngIf` based on `isSignedIn` that installs a router-outlet; is that correct? If so, I only know this because I recently looked at the code, it's not super obvious here otherwise. Can you add a comment?

Reply:
Yeah. I'm also not entirely certain if `this.router.url` is the real URL, or the url of the router?